src/server/database/Database/Field.cpp

```diff
@@ -1,57 +1,240 @@
 /*
- * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>
- * Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
- * Copyright (C) 2005-2009 MaNGOS <http://getmangos.com/>
+ * Copyright (C) 2016+ AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license, you may redistribute it and/or modify it under version 2 of the License, or (at your option), any later version.
+ * Copyright (C) 2021+ WarheadCore <https://github.com/WarheadCore>
  */
 
 #include "Field.h"
 #include "Errors.h"
+#include "Log.h"
+#include "MySQLHacks.h"
 
 Field::Field()
 {
     data.value = nullptr;
-    data.type = MYSQL_TYPE_NULL;
     data.length = 0;
     data.raw = false;
+    meta = nullptr;
 }
 
-Field::~Field()
+Field::~Field() = default;
+
+uint8 Field::GetUInt8() const
 {
-    CleanUp();
+    if (!data.value)
+        return 0;
+
+#ifdef WARHEAD_STRICT_DATABASE_TYPE_CHECKS
```

Comment:
Warhead? Another core to make Azerothcore even more confusing?

Reply:
my bad